### PR TITLE
Add dynamic import note regarding babel preset env

### PR DESCRIPTION
**Pull request by nathanjisaac**
Add simple note about dynamic import plugin and the preset env plugin. 

Requested in: https://github.com/babel/babel/issues/7402

### CODE REVIEW

File: docs/plugin-syntax-dynamic-import.md [hunk 1]
```diff
@@ -36,3 +36,28 @@ require("@babel/core").transform("code", {
 });
 ```
 
+## Dynamic imports with babel-preset-env
+
+The babel-preset-env plugin doesn't know that the dynamic import plugin with translate `import` into a `Promise`. For example, IE requires the `promise` and `iterator` polyfill to be added manually.
```
Comment (tjefferson08) [inline docs/plugin-syntax-dynamic-import.md]: >that the dynamic import plugin with translate

Typo: plugin _will_ translate

Also, these docs would be very helpful on the site!

File: website/versioned_docs/version-7.0.0/plugin-syntax-dynamic-import.md [hunk 2]
```diff
@@ -37,3 +37,28 @@ require("@babel/core").transform("code", {
 });
 ```
 
+## Dynamic imports with babel-preset-env
+
+The babel-preset-env plugin doesn't know that the dynamic import plugin with translate `import` into a `Promise`. For example, IE requires the `promise` and `iterator` polyfill to be added manually.
```
Comment (tjefferson08) [inline website/versioned_docs/version-7.0.0/plugin-syntax-dynamic-import.md]: _will_ translate `import` (same typo)

**Comment by nathanjisaac** [comment]
@tjefferson08 I fixed the typo but the `git commit` hook required me to add a new 7.1.0 document. Let me know if anything else needs to change.

**Comment by existentialism** [comment]
@nathanjisaac I made a slight tweak (and change for versioning, docusaurus' versioning isn't totally obvious :D)

Thanks for the PR!

**Pull request merged**